depth-test_z-acc_spatial-noise/depth_test.py

```diff
@@ -0,0 +1,174 @@
+import open3d as o3d
+import numpy as np
+from utils import *
+
+class DepthTest:
+	def __init__(self):
+		self.camera_dir = np.array([0, 0, -1])
+		self.plane_normal = np.array([0, 0, -1])
+		self.camera_wall_distance = 1 # m
+		self.plane_distance = -self.camera_wall_distance
+		self.plane_coeffs = (0,0,-self.camera_wall_distance)
+
+		self.point_cloud_corrected = o3d.geometry.PointCloud()
+		self.plane_fit_corrected_pcl = o3d.geometry.PointCloud()
+		self.plane_fit_pcl = o3d.geometry.PointCloud()
+		self.plane_fit_visualization = False
+
+		self.fitted = False
+
+		self.z_accuracy_medians = []
+		self.z_means = []
+		self.spatial_noise_rmses = []
+		self.samples = 0
+
+	def set_ground_truth(self, point_cloud: o3d.geometry.PointCloud):
+		points = np.asarray(point_cloud.points)
+		G = np.ones_like(points)
+		G[:, 0] = points[:, 0]
+		G[:, 1] = points[:, 1]
+		Z = points[:, 2]
+
+		self.camera_wall_distance = -np.mean(Z)
```

Comment:
Can you cross check if mean or median would be best here by some trial and errors ?

Reply:
I have a feeling median might do better.

Reply:
It is hard to tell which is better but the median appears more consistent over multiple measurements so I'll change it to median. The difference between mean and meadian is less than a millimeter at one meter distance.